app/controllers/mock.php

```diff
@@ -646,6 +649,69 @@
+    ->param('installation_id', '', new Text(256, 0), 'GitHub installation ID', true)
+    ->param('projectId', '', new Text(2048), 'Project ID of the project where app is to be installed', true)
+    ->inject('gitHub')
+    ->inject('project')
+    ->inject('response')
+    ->inject('dbForConsole')
+    ->action(function (string $providerInstallationId, string $projectId, GitHub $github, Document $project, Response $response, Database $dbForConsole) {
+        if (empty($projectId)) {
+            $error = 'Installation requests from organisation members for the Appwrite GitHub App are currently unsupported. To proceed with the installation, login to the Appwrite Console and install the GitHub App.';
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, $error);
+        }
+
+        $project = $dbForConsole->getDocument('projects', $projectId);
+
+        if ($project->isEmpty()) {
+            $error = 'Project with the ID from state could not be found.';
+            throw new Exception(Exception::PROJECT_NOT_FOUND, $error);
+        }
+
+        // Create / Update installation
+        if (!empty($providerInstallationId)) {
+            $privateKey = App::getEnv('_APP_VCS_GITHUB_PRIVATE_KEY');
+            $githubAppId = App::getEnv('_APP_VCS_GITHUB_APP_ID');
+            $github->initializeVariables($providerInstallationId, $privateKey, $githubAppId);
+            $owner = $github->getOwnerName($providerInstallationId) ?? '';
+
+            $projectInternalId = $project->getInternalId();
+
+            $teamId = $project->getAttribute('teamId', '');
+
+            $installation = new Document([
+                '$id' => ID::unique(),
+                '$permissions' => [
+                    Permission::read(Role::team(ID::custom($teamId))),
+                    Permission::update(Role::team(ID::custom($teamId), 'owner')),
+                    Permission::update(Role::team(ID::custom($teamId), 'developer')),
+                    Permission::delete(Role::team(ID::custom($teamId), 'owner')),
+                    Permission::delete(Role::team(ID::custom($teamId), 'developer')),
+                ],
+                'providerInstallationId' => $providerInstallationId,
+                'projectId' => $projectId,
+                'projectInternalId' => $projectInternalId,
+                'provider' => 'github',
+                'organization' => $owner,
+                'personal' => false
+            ]);
+
+            $installation = $dbForConsole->createDocument('installations', $installation);
+        } else {
+            $error = 'Installation of the Appwrite GitHub App on organization accounts is restricted to organization owners. As a member of the organization, you do not have the necessary permissions to install this GitHub App. Please contact the organization owner to create the installation from the Appwrite console.';
+
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, $error);
+        }
```

Comment:
Same as comment above. If you remove `true` from `param()` definition, you don't need this empty check and exception.